lib/lbt/analyzer/analyzeLibraryJS.js

```diff
@@ -82,6 +82,60 @@ async function analyze(resource) {
 	return libInfo;
 }
 
+const initLibraryDefintions = [];
+/**
+ * Determines whether the node is a initLibrary call.
+ *
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isInitLibraryCall(node) {
+	// sap.ui.getCore()
+	if (
+		node.type == Syntax.CallExpression &&
+		node.callee.type === Syntax.MemberExpression &&
+		isMethodCall(node.callee.object, CALL__SAP_UI_GETCORE) &&
+		isIdentifier(node.callee.property, "initLibrary") &&
+		node.arguments.length === 1 &&
+		node.arguments[0].type === Syntax.ObjectExpression
+	) {
+		return true;
+	}
+
+	// Find whether there's potential initLibrary() dependency in AMD i.e. "sap/ui/core/Core"
+	if (node.type === Syntax.Program && !initLibraryDefintions.length) {
+		const amdDepDefinitions = node.body[0]?.expression?.arguments[0]?.elements;
+		const amdFnArguments = node.body[0]?.expression?.arguments[1]?.params || [];
+
+		for (let i = 0; i < amdFnArguments.length; i++) {
+			const fnArgName = amdFnArguments[i].name; // Name of the variable
+			const amdDef = amdDepDefinitions[i].value;
+			if (AMD__LIB_INIT_RESOURCES[amdDef]) {
+				// Add just the ones that are available
+				initLibraryDefintions.push([fnArgName, AMD__LIB_INIT_RESOURCES[amdDef]]);
+			}
+		}
+	}
```

Comment:
I expect that there's a better solution than having a side-effect within this `isInitLibraryCall` function.
Also, wouldn't it make sense to use `lbt/calls/SapUiDefine.js` to re-use the analysis of the sap.ui.define call?

Reply:
Thanks. Switched to sapUiDefine analyzer